client/app/components/project/package-section.hbs

```diff
@@ -0,0 +1,92 @@
+<div ...attributes>
+  <h2>{{@packageType}}</h2>
+
+  <div class="grid-x grid-padding-x">
+    <div class="cell large-6">
+      <ol class="no-bullet">
+        {{#each @packages as |package|}}
+          <Project::PackageListItem
+            @package={{package}}
+          />
+        {{/each}}
+      </ol>
+    </div>
+    <div class="cell large-6">
+      {{#let (flatten (map-by 'invoices' @packages)) as |invoices|}}
+        {{#if invoices}}
+          <h4>Invoices</h4>
+
+          {{#each invoices as |invoice|}}
+            <Project::PackageSection::InvoiceInfo
+              @invoice={{invoice}}
+            />
+          {{/each}}
+
+          <br/>
+          <br/>
+          <button
+            type="button"
+            class="secondary button"
+            {{on 'click' (fn this.beginPayment)}}
+          >
+            Pay
+          </button>
+
+          <Ui::GenericModal
+            @show={{this.isPaying}}
+            @closeButtonAction={{fn (mut this.isPaying) false}}
+          >
+            <h3>Invoices to be paid</h3>
+
+            {{#each invoices as |invoice|}}
+              <Project::PackageSection::InvoiceInfo
+                @invoice={{invoice}}
+              />
+            {{/each}}
+
+            <br/>
+            <br/>
+            <h3>Total: ${{reduce (fn this.sum) 0 (map-by 'dcpGrandtotal' invoices)}}</h3>
```

Comment:
would this be more clear if it was all done in a getter/cp?

Reply:
Just for the sake of discussion on handlebars helpers, can you try to make a guess at what this does?

Reply:
It's basically a reduce function and you're supplying the accumulator function (simple add). It just looks a tiny bit wonky at first and could probably be done in a getter 

So it's adding up each dcpGrandtotal value on each invoice